From a00fa3e82175403a41f53268c1932205725bf9d8 Mon Sep 17 00:00:00 2001
From: Saleem Abdulrasool <compnerd@compnerd.org>
Date: Tue, 29 Nov 2011 19:19:59 -0800
Subject: [PATCH] web view: fix icon names

The GTK+ 3.0 themes have a different name for the dialog icons
(dialog-{information,error}) rather than the stock GTK+ icon names
(gtk-dialog-{info,error}).  Use the proper names.

Additionally, ensure that icon_info is not NULL when performing the lookup,
which will fail.  This part of the change is to avoid a possible crash/segfault
caused by the lookup without a valid icon_info.

Signed-Off-By: Saleem Abdulrasool <compnerd@compnerd.org>
---
 embed/ephy-web-view.c |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/embed/ephy-web-view.c b/embed/ephy-web-view.c
index cc5e61e..f16bd19 100644
--- a/embed/ephy-web-view.c
+++ b/embed/ephy-web-view.c
@@ -2044,7 +2044,7 @@ ephy_web_view_load_error_page (EphyWebView *view,
       button_label = g_strdup (_("Try again"));
 
       html_file = ephy_file ("error.html");
-      stock_icon = GTK_STOCK_DIALOG_ERROR;
+      stock_icon = "dialog-error";
       break;
     case EPHY_WEB_VIEW_ERROR_PAGE_CRASH:
       page_title = g_strdup_printf (_("Oops! Error loading %s"), hostname);
@@ -2062,7 +2062,7 @@ ephy_web_view_load_error_page (EphyWebView *view,
       button_label = g_strdup (_("Load again anyway"));
 
       html_file = ephy_file ("recovery.html");
-      stock_icon = GTK_STOCK_DIALOG_INFO;
+      stock_icon = "dialog-information";
       break;
     default:
       return;
@@ -2075,7 +2075,7 @@ ephy_web_view_load_error_page (EphyWebView *view,
                                           48,
                                           GTK_ICON_LOOKUP_GENERIC_FALLBACK);
 
-  image_data = get_file_content_as_base64 (gtk_icon_info_get_filename (icon_info));
+  image_data = icon_info ? get_file_content_as_base64 (gtk_icon_info_get_filename (icon_info)) : NULL;
 
   g_file_get_contents (html_file, &template, NULL, NULL);
 
@@ -2087,7 +2087,7 @@ ephy_web_view_load_error_page (EphyWebView *view,
                    ((gtk_widget_get_default_direction () == GTK_TEXT_DIR_RTL) ? "rtl" : "ltr"),
                    page_title,
                    uri,
-                   image_data,
+                   image_data ? image_data : "",
                    msg_title, msg, button_label);
 
   g_free (template);
-- 
1.7.7.4

